packages/flutter/lib/src/rendering/debug_overflow_indicator.dart

```diff
@@ -0,0 +1,301 @@
+    String overflowText = '';
+    assert(overflows.isNotEmpty,
+        "Somehow $runtimeType didn't actually overflow like it thought it did.");
+    switch (overflows.length) {
+      case 1:
+        overflowText = overflows.first;
+        break;
+      case 2:
+        overflowText = '${overflows.first} and ${overflows.last}';
+        break;
+      default:
+        overflows[overflows.length - 1] = 'and ${overflows[overflows.length - 1]}';
+        overflowText = overflows.join(', ');
+    }
+    FlutterError.reportError(
+      new FlutterErrorDetailsForRendering(
+        exception: 'A $runtimeType overflowed by $overflowText.',
+        library: 'rendering library',
+        context: 'during layout',
+        renderObject: this,
+        informationCollector: (StringBuffer information) {
+          information.writeln(overflowHints);
+          information.writeln('The specific $runtimeType in question is:');
+          information.writeln('  ${toStringShallow(joiner: '\n  ')}');
+          information.writeln('◢◤' * (FlutterError.wrapWidth ~/ 2));
+        },
+      ),
+    );
+  }
+
+  /// To be called when the overflow indicators should be painted.
+  ///
+  /// Typically only called if there is an overflow, and only from within a
+  /// debug build.
+  ///
+  /// See example code in [DebugOverflowIndicatorMixin] documentation.
+  void paintOverflowIndicator(
+      PaintingContext context, Offset offset, Rect containerRect, Rect childRect,
+      {String overflowHints}) {
+    final RelativeRect overflow = new RelativeRect.fromRect(containerRect, childRect);
+
+    if (overflow.left <= 0.0 &&
+        overflow.right <= 0.0 &&
+        overflow.top <= 0.0 &&
+        overflow.bottom <= 0.0) {
+      return;
+    }
+
+    final List<_OverflowRegionData> overflowRegions = _calculateOverflowRegions(overflow, containerRect);
+    for (_OverflowRegionData region in overflowRegions) {
+      context.canvas.drawRect(region.rect.shift(offset), _indicatorPaint);
+
+      _indicatorLabel.text = new TextSpan(
+        text: region.label,
+        style: _indicatorTextStyle,
+      );
+      _indicatorLabel.layout();
+
+      final Offset labelOffset = region.labelOffset + offset;
+      final Offset centerOffset = new Offset(-_indicatorLabel.width / 2.0, 0.0);
+      final Rect textBackgroundRect = centerOffset & _indicatorLabel.size;
+      context.canvas.save();
+      context.canvas.translate(labelOffset.dx, labelOffset.dy);
+      context.canvas.rotate(region.rotation);
+      context.canvas.drawRect(textBackgroundRect, _labelBackgroundPaint);
+      _indicatorLabel.paint(context.canvas, centerOffset);
+      context.canvas.restore();
+    }
+
+    if (_overflowReportNeeded) {
+      _overflowReportNeeded = false;
+      _reportOverflow(overflow, overflowHints);
+    }
+  }
+
+  @override
+  void reassemble() {
+    super.reassemble();
+    // Each time we reassemble, we want to see the next overflow debug message.
```

Comment:
this comment doesn't really add much (it's just describing the code).

If you want a comment here, maybe something like "Users expect error messages to be shown again after hot reload" or something.